app/services/HCAService.js

```diff
@@ -0,0 +1,264 @@
+   *
+   * @returns {boolean}
+   */
+  async hasSavedAuthorities() {
+    const authorities = await this.getUserAuthorityList();
+    return authorities && authorities.length > 0;
+  }
+
+  /**
+   * Alerts a user that there are new Healthcare Authorities in their region.
+   * Includes information on the number of Authorities in their current location.
+   *
+   * @param {numAuthorities} Number number of new authorities
+   * @returns {void}
+   */
+  async pushAlertNewAuthoritesFromLoc(numAuthorities) {
+    PushNotification.localNotification({
+      title: languages.t('label.authorities_new_in_area_title'),
+      message: languages.t('label.authorities_new_in_area_msg', {
+        count: numAuthorities,
+      }),
+    });
+  }
+
+  /**
+   * Alerts a user that they have been subscribed to new Healthcare Authorities
+   * in their region. Includes information on the number of Authorities in
+   * their current location.
+   *
+   * @param {numAuthorities} Number number of new authorities
+   * @returns {void}
+   */
+  async pushAlertNewSubscribedAuthorities(numAuthorities) {
+    PushNotification.localNotification({
+      title: languages.t('label.authorities_new_subcription_title', {
+        count: numAuthorities,
+      }),
+      message: languages.t('label.authorities_new_subcription_msg', {
+        count: numAuthorities,
+      }),
+    });
+  }
+
+  /**
+   * Returns the `url` value for an authority
+   * @param {authority} Authority Healthcare Authority object
+   * @returns {string}
+   */
+  getAuthorityUrl(authority) {
+    const authorityName = Object.keys(authority)[0];
+    const urlKey = authority[authorityName][0];
+    return urlKey && urlKey['url'];
+  }
+
+  /**
+   * Returns the `bounds` value for an authority
+   * @param {authority} Authority Healthcare Authority object
+   * @returns {Object} contains a `bounds` key.
+   * A `bounds` contains a `ne` and `sw` object that each
+   * have a valid GPS point containing `longitude` and `latitude` keys.
+   */
+  getAuthorityBounds(authority) {
+    const authorityName = Object.keys(authority)[0];
+    const boundsKey = authority[authorityName][1];
+    return boundsKey && boundsKey['bounds'];
+  }
+
+  /**
+   * Checks if a given point is inside the bounds of the given authority
+   * @param {point} Object contains a `latitude` and `longitude` field
+   * @param {authority} Authority Healthcare Authority object
+   * @returns {boolean}
+   */
+  isPointInAuthorityBounds(point, authority) {
+    const locHelper = new LocationData();
+    const bounds = this.getAuthorityBounds(authority);
+
+    return bounds && locHelper.isPointInBoundingBox(point, bounds);
+  }
+
+  /**
+   * Iterates over the full list of authorities and checks
+   * if there is any GPS point in the user's full 28-day location history
+   * that is within the bounds of the authority.
+   *
+   * @returns {Array} List of health care authorities
```

Comment:
`@returns {Type[]}`

Reply:
Could use your feedback on the updates I made here.